scm-webapp/src/main/java/sonia/scm/api/v2/resources/RepositoryPermissionToRepositoryPermissionDtoMapper.java

```diff
@@ -51,18 +53,19 @@
   @Mapping(target = "attributes", ignore = true) // We do not map HAL attributes
   public abstract RepositoryPermissionDto map(RepositoryPermission permission, @Context Repository repository);
 
+  @Mapping(target = "attributes", ignore = true) // We do not map HAL attributes
+  public abstract RepositoryPermissionDto map(RepositoryPermission permission, @Context Namespace namespace);
 
   @BeforeMapping
   void validatePermissions(@Context Repository repository) {
     RepositoryPermissions.permissionRead(repository).check();
   }
 
-  /**
-   * Add the self, update and delete links.
-   *
-   * @param target     the mapped dto
-   * @param repository the repository
-   */
+  @BeforeMapping
+  void validatePermissions(@Context Namespace namespace) {
+    NamespacePermissions.permissionRead().check();
+  }
+
```

Comment:
Should the mapper do to the permission check? I thought it should happen earlier. Even the resource seems a better place for this.

Reply:
Right. This one was a "copy" of the existing logic, though it is not necessary here. Will remove this.